common/membership/hashring_test.go

```diff
@@ -148,13 +163,12 @@ func TestRefreshWillNotifySubscribers(t *testing.T) {
 		HostsRemoved: []string{"c"},
 	}
 
-	hr := newHashring("test-service", pp, log.NewNoop(), metrics.NoopScope(0))
-	hr.Start()
+	td.hashRing.Start()
```

Comment:
let's stop this and also validate no goroutines leaking using `goleak.VerifyNone(t)`

Reply:
also in other tests if this was missed

Reply:
Added to all tests via `Cleanup()`